pkg/apis/nmstate/v1alpha1/nodenetworkconfigurationpolicy_types.go

```diff
@@ -0,0 +1,74 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// NodeNetworkConfigurationPolicyMatch define the matching criteria to apply
+// the policy
+// +k8s:openapi-gen=true
+type NodeNetworkConfigurationPolicyMatch struct {
```

Comment:
in the original design doc, Match was only for interface matching, nodeSelector was under Spec directly. I'd be in favor of that, to avoid too much nesting.

Reply:
Like k8s https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/types.go#L2878

Reply:
Don't understand your point here. nodeSelector is a map that should live in the top scope of Spec. `NodeNetworkConfigurationPolicyMatch` you define here is not needed in this PR, `match` should be later used to match interfaces running on a node.